include/mqtt/endpoint.hpp

```diff
@@ -8954,18 +8945,21 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
             clean_session_ = connect_flags::has_clean_session(info.connect_flag);
 
             buf.remove_prefix(info.header_len); // consume buffer
-            process_connect_impl(
-                force_move(session_life_keeper),
-                force_move(buf),
-                [&] {
-                    if (version_ == protocol_version::v5) {
-                        return connect_phase::properties;
-                    }
-                    return connect_phase::client_id;
-                } (),
-                force_move(info),
-                force_move(self)
-            );
+            if(version_ == protocol_version::v5) {
+                process_connect_impl<connect_phase::properties>(
+                    force_move(session_life_keeper),
+                    force_move(buf),
+                    force_move(info),
+                    force_move(self)
+                );
+            } else {
```

Comment:
```suggestion
            } 
            else {
```

Please use this style.
See https://github.com/redboltz/mqtt_cpp/wiki/Coding-Rules

Could you update other `else` and `else if` ?